models/events.go

```diff
@@ -0,0 +1,121 @@
+package models
+
+import (
+	gethTypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/onflow/cadence"
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/flow-go-sdk"
+	"github.com/onflow/flow-go/fvm/evm/types"
+	"golang.org/x/exp/slices"
+)
+
+var evmLocation common.Location = flow.EVMLocation{}
+
+var BlockExecutedEventType = evmLocation.TypeID(
+	nil,
+	string(types.EventTypeBlockExecuted),
+)
+
+var TransactionExecutedEventType = evmLocation.TypeID(
+	nil,
+	string(types.EventTypeTransactionExecuted),
+)
+
+// isBlockExecutedEvent checks whether the given event contains block executed data.
+func isBlockExecutedEvent(event cadence.Event) bool {
+	if event.EventType == nil {
+		return false
+	}
+	return common.TypeID(event.EventType.ID()) == BlockExecutedEventType
+}
+
+// isTransactionExecutedEvent checks whether the given event contains transaction executed data.
+func isTransactionExecutedEvent(event cadence.Event) bool {
+	if event.EventType == nil {
+		return false
+	}
+	return common.TypeID(event.EventType.ID()) == TransactionExecutedEventType
+}
+
+type CadenceEvents struct {
+	events flow.BlockEvents
+}
+
+func NewCadenceEvents(events flow.BlockEvents) *CadenceEvents {
+	return &CadenceEvents{events: events}
+}
+
+// Blocks finds the block evm events and decodes it into the blocks slice,
+// if no block events are found nil slice is returned.
+//
+// Return values:
+// blocks, nil - if blocks are found
+// nil, nil - if no block are found
```

Comment:
What is the use of returning `nil, nil` :thinking: 
Below we use something like:
```go
blocks, err := events.Blocks()
if err != nil {
	return err
}
for _, block := range blocks {
	if err := e.indexBlock(events.CadenceHeight(), block); err != nil {
		return err
	}
}
```
We are not checking whether `blocks` is `nil`. Does `range` play nicely with that?
The same we are doing for `Transactions()`.

Reply:
forgot to update yeah, it supposed to be empty slice, but also if nil range plays nice

Reply:
Interesting, I wasn't aware of that :+1: